### PR TITLE
#2599 TemperatureSync to navigation bar

### LINKED ISSUE #2599: Add TemperatureSync to NavigationBar
## Is your feature request related to a problem? Please describe.

Need a way to open the temperature sync modal and to show the current progress.

## Describe the solution you'd like

Add the temperature sync status just like regular sync

## Implementation

N/A

## Describe alternatives you've considered

N/A

## Additional context

Not sure what the icon should be.......

image

??

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #2599 

## Change summary

- Adds an icon for temperature syncing to the navigation bar
- Adds logic to open the temperature syncing modal
- Adds the actual temperature syncing modal!

## Testing

N/A

### Related areas to think about

Just figured out how to screen shot on a tablet 🤣 
[`adb exec-out screencap -p > ~/.../picname.png`]!

temperaturesyncicon
temperaturesyncing

The icon is pretty naff but it'll be super easy to just change icons if we want!

### CODE REVIEW

File: src/widgets/TemperatureSyncState.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  return { syncMessage, isSyncing };
+};
+
+const mapDispatchToProps = dispatch => {
+  const onOpenModal = () => dispatch(TemperatureSyncActions.openModal());
+  return { onOpenModal };
+};
+
+export const TemperatureSyncStateComponent = ({ isSyncing, onOpenModal, syncMessage }) => (
```
Comment (wlthomson) [inline src/widgets/TemperatureSyncState.js:33]: Pure?

File: src/widgets/icons.js [hunk 2]
```diff
@@ -126,3 +126,13 @@ export const WifiIcon = ({ color, size, style }) => (
 );
 WifiIcon.defaultProps = { color: DARK_GREY, size: 22, style: { marginLeft: 10 } };
 WifiIcon.propTypes = { color: PropTypes.string, size: PropTypes.number, style: PropTypes.object };
+
+export const TemperatureIcon = ({ color, size, style }) => (
```
Comment (wlthomson) [inline src/widgets/icons.js:130]: Pure 😆 ?

File: src/widgets/TemperatureSyncIcon.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/TemperatureSyncIcon.js]: you're 2020 6a4ed2b22e35aec3fddc9a09403419bdb676c1f4

File: src/widgets/TemperatureSyncIcon.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { GREY, DARK_GREY } from '../globalStyles';
+import { CloudIcon, ArrowIcon, TemperatureIcon } from './icons';
+
+export const TemperatureSyncIcon = ({ isActive }) => {
+  const iconColor = isActive ? DARK_GREY : GREY;
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/TemperatureSyncIcon.js:13]: 4e76d48554118e8bf5d0a97e49d0f5ed1455fb06

File: src/widgets/HeaderRight.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import { FlexRow } from './FlexRow';
+import { TemperatureSyncState } from './TemperatureSyncState';
+
+const mapStateToProps = state => {
+  const usingVaccines = selectUsingVaccines(state);
+
+  return { usingVaccines };
+};
+
+const HeaderRightComponent = ({ usingVaccines }) => (
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/HeaderRight.js:23]: this ones wrapped in `connect`, which is by default made a pure component. There are other options you can add to `connect` to specify even more behaviour too: https://react-redux.js.org/api/connect#options-object

File: src/widgets/TemperatureSyncState.js (hunk 1 above)
Comment (joshxg) [reply src/widgets/TemperatureSyncState.js:33]: As above don't need to when you `connect`

File: src/widgets/icons.js (hunk 2 above)
Comment (joshxg) [reply src/widgets/icons.js:130]: pretty much all the icons are non-memoized - it should be consistent though!

**Pull request merged**